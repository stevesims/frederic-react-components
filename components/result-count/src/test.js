import React from 'react';
import { mount, shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import ResultCount from '.';

expect.extend(createMatchers(emotion));

describe('ResultCount', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ResultCount />);
    expect(wrapper.exists()).toBe(true);
  });

  it('renders with count', () => {
    const wrapper = shallow(<ResultCount>000</ResultCount>);
    expect(wrapper.childAt(0).text()).toBe('000');
  });

  describe('applies styles in response to props', () => {
    it('sets color from prop', () => {
      const wrapper = shallow(<ResultCount color="blue">000</ResultCount>);
      expect(wrapper).toHaveStyleRule('color', 'blue');
    });

    it('sets background from prop', () => {
      const wrapper = shallow(<ResultCount backgroundColor="yellow">000</ResultCount>);
      expect(wrapper).toHaveStyleRule('background', 'yellow');
    });
  });

  it('matches snapshot', () => {
    const wrapper = mount(<ResultCount>0</ResultCount>);
    expect(wrapper).toMatchSnapshot();
  });
});

