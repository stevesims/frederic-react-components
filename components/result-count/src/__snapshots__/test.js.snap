// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCount matches snapshot 1`] = `
.emotion-0 {
  background: #6f777b;
  color: #ffffff;
  display: inline-block;
  font-family: Roboto,Arial,sans-serif;
  font-size: 15px;
  line-height: 1.25;
  min-height: 20px;
  min-width: 20px;
  padding: 1px 4px;
  text-align: center;
}

<ResultCount
  backgroundColor="#6f777b"
  color="#ffffff"
>
  <StyledResultCount
    backgroundColor="#6f777b"
    color="#ffffff"
  >
    <span
      className="emotion-0 emotion-1"
      color="#ffffff"
    >
      0
    </span>
  </StyledResultCount>
</ResultCount>
`;
